elide-async/src/test/java/com/yahoo/elide/async/service/AsyncExecutorServiceTest.java

```diff
@@ -72,25 +66,33 @@ public void testAsyncExecutorServiceSet() {
         assertNotNull(service.getRunners());
         assertEquals(60, service.getMaxRunTime());
         assertNotNull(service.getExecutor());
-        assertNotNull(service.getInterruptor());
+        assertNotNull(service.getUpdater());
         assertEquals(asyncQueryDao, service.getAsyncQueryDao());
     }
 
+    //Test for regular query execution before asyncAfterSeconds expires
     @Test
-    public void testExecuteQuery() {
+    public void testExecuteQuery() throws ExecutionException, TimeoutException, InterruptedException {
 
-        service.executeQuery(queryObj, testUser, NO_VERSION);
-        verify(asyncQueryDao, times(0)).updateStatus(queryObj.getId(), QueryStatus.QUEUED);
+       when(queryObj.getAsyncAfterSeconds()).thenReturn(10);
+       service.executeQuery(queryObj, testUser, NO_VERSION);
+       verify(queryObj, times(1)).setStatus(QueryStatus.PROCESSING);
```

Comment:
Should we also verify
 verify(asyncQueryDao, times(1)).updateAsyncQueryResult(queryResultObj, query.getId());